File: InMotionGIT.Common - Copy/MetadataCatalogAttributes/ElementBehaviorAttribute.vb
Namespace Attributes

    ''' <summary>
    ''' Define las características de tamaño, precisión y/o escala dependiendo del tipo asociado a la propiedad.
    ''' </summary>
    ''' <remarks></remarks>
    <AttributeUsage(AttributeTargets.Property)>
    Public NotInheritable Class ElementBehaviorAttribute
        Inherits Attribute

        Private _Scale As Integer
        Private _Size As Integer
        Private _Precision As Integer
        Private _formControl As Enumerations.EnumFormControls

        Public ReadOnly Property Scale() As Integer
            Get
                Return _Scale
            End Get
        End Property

        Public ReadOnly Property Size() As Integer
            Get
                Return _Size
            End Get
        End Property

        Public ReadOnly Property FormControl() As Enumerations.EnumFormControls
            Get
                Return _formControl
            End Get
        End Property

        Public ReadOnly Property Precision() As Integer
            Get
                Return _Precision
            End Get
        End Property

        Public Sub New(ByVal size As Integer)
            _Size = size
            _Precision = size
        End Sub

        Public Sub New(ByVal precision As Integer, ByVal scale As Integer)
            _Scale = scale
            _Size = precision
            _Precision = precision
        End Sub

        Public Sub New(ByVal formControl As Enumerations.EnumFormControls)
            _formControl = formControl
        End Sub

    End Class

End Namespace

File: Common/InMotionGIT.Common.Domain/Attributes/ElementBehaviorAttribute.cs
using System;

namespace InMotionGIT.Common.Domain.Attributes;


/// <summary>
/// Define las características de tamaño, precisión y/o escala dependiendo del tipo asociado a la propiedad.
/// </summary>
/// <remarks></remarks>
[AttributeUsage(AttributeTargets.Property)]
public sealed class ElementBehaviorAttribute : Attribute
{

    private int _Scale;
    private int _Size;
    private int _Precision;
    private Enumerations.EnumFormControls _formControl;

    public int Scale
    {
        get
        {
            return _Scale;
        }
    }

    public int Size
    {
        get
        {
            return _Size;
        }
    }

    public Enumerations.EnumFormControls FormControl
    {
        get
        {
            return _formControl;
        }
    }

    public int Precision
    {
        get
        {
            return _Precision;
        }
    }

    public ElementBehaviorAttribute(int size)
    {
        _Size = size;
        _Precision = size;
    }

    public ElementBehaviorAttribute(int precision, int scale)
    {
        _Scale = scale;
        _Size = precision;
        _Precision = precision;
    }

    public ElementBehaviorAttribute(Enumerations.EnumFormControls formControl)
    {
        _formControl = formControl;
    }

}